### PR TITLE
#165305265 Users should be able to create their profiles

### PR DESCRIPTION
**Pull request by Siffersari**
### What does this PR do?

Adds endpoints responsible for creating and updating user profiles

### Description of the task to be completed?

- Add endpoint responsible for creating a user profile
- Add endpoint responsible for fetching a user profile
- Add endpoint responsible for updating a user profile
- Add tests for creating, fetching and updating a user profile

### How should you manually test this?

- [x] **Clone the repository**

        $ git clone https://github.com/andela/ah-the-jedi-backend.git

- [x] **Switch to testing branch**

       $ git checkout ft-user-create-profile-165305265 

- [x] **Switch to the ah-the-jedi-backend directory**

       $ cd ah-the-jedi-backend/

 

- [x] **Create and activate the virtual environment**

      $  virtualenv -p python .venv

      $ source .venv/bin/activate

- [x] **Install project dependencies**

       $ pip install -r requirements.txt

- [x] **Run the database migrations**

       $ python manage.py migrate

### Running the application

      $ python manage.py runserver

- [x] **Signup and activate user**

           POST <your-localhost>/api/signup/

           Creates a user profile on successful signup

- [x] **Login user**

       POST <your-localhost>/api/login/ 

       Use the token provided after login for authorization 

  
- [x] **Fetch a user  profile**

      GET <your-localhost>/api/profiles/<username>

- [x] **Update a user  profile (with and without image)**

      PUT <your-localhost>/api/profiles/<username>

      Authentication required

### Testing

      $ python manage.py test

### Prerequisites

Python, Git,  Virtualenv

### What are the relevant PT stories?

[#165305265](https://www.pivotaltracker.com/story/show/165305265)

### Screenshots

***Successful Profile Fetch***

Screenshot 2019-04-30 at 12 32 58

***Successful Profile Update***
Screenshot 2019-04-30 at 12 30 56

***JWT Authentication handling***
Screenshot 2019-04-30 at 12 32 41

***Validations***
Screenshot 2019-04-30 at 12 31 54
Screenshot 2019-04-30 at 12 31 30

### CODE REVIEW

**Review by Benkimeric — APPROVED** [review]
Looks good to me

**Review by ElishaMisoi — APPROVED** [review]

**Review by katherine95 — APPROVED** [review]

**Review by AlvinMugambi — APPROVED** [review]
Great implementation

File: authors/apps/profiles/models.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+from django.db import models
+from ..authentication.models import User
+from django.db.models.signals import post_save
+
+
+# Sample request data
```
Comment (rnjane) [inline authors/apps/profiles/models.py]: @Siffersari remove all the unnecessary comments

**Review by rnjane — CHANGES_REQUESTED** [review]

File: authors/apps/profiles/models.py (hunk 1 above)
Comment (Siffersari) [reply authors/apps/profiles/models.py]: Thank you for that. I have removed the comments

**Pull request merged**